test/pinecone_test.dart

```diff
@@ -48,7 +48,7 @@ void main() async {
   // TEST: Create Index
   // ==========================================
 
-  test('Create Index', () async {
+  test('Create Index', timeout: Timeout(Duration(minutes: 1)), () async {
```

Comment:
@davidmigloz can you undo these changes to the tests.

In my testing, the Pinecone database can be very inconsistent with the response times and a timeout of 1 minute is definitely not enough. In fact there are wait durations in this test method that are at least that long. 

Even when the index is "ready" it still might not accept upsert requests until a few seconds later, sometimes even 10s of seconds later. Some weird quirks of the Pinecone DB API. To protect against this, there is an await at the very end of this test, just in case:

```dart
await Future.delayed(Duration(seconds: 60));
``` 

The other changes look good and match the auto-generated code. I'll get this merged in and published after that!

Reply:
I actually added it because the test was timing out (the [default timeout](https://pub.dev/packages/test#timeouts) for dart tests is 30 seconds and it was taking longer than that). 

Maybe we can increase the timeout globally for the file and make it more than 1min.
```dart
@Timeout(Duration(minutes: 5))
```

The other delays I added were because the `describeIndexStats` endpoint wasn't replying up-to-date stats if you were calling it just after modifying the index. So the tests were failing.

Reply:
> Maybe we can increase the timeout globally for the file and make it more than 1min.

Sounds good. Let's do that. FWIW, the timeout is actually set in the `make test` execution. Probably good to formally add it to the script itself.

https://github.com/tazatechnology/pinecone/blob/8f0237702890512a7e8748678e2616895ac11cbf/Makefile#L34-L44

Reply:
I typically use `make test` to do all the analysis and testing for the repo, which includes `dart test` with custom command line args. As seen above. Probably why you were running into the timeout when you ran it manually.

Reply:
ahh I missed that the timeout was in the Makefile as well, sorry!

I've made it global to the file and increased it to 10min to match the one from the makefile.